includes/migrations/class-track-upe-status.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Class Track_Upe_Status
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Migrations;
+
+defined( 'ABSPATH' ) || exit;
+
+use WCPay\Tracker;
+
+/**
+ * Class Track_Upe_Status
+ *
+ * Fires an event on plugin upgrade to track whether UPE is enabled.
+ * Runs only once. We want to know whether existing install had it
+ * enabled before the current version, when we started to track it.
+ */
+class Track_Upe_Status {
+
+	const IS_TRACKED_OPTION = 'wcpay_upe_is_tracked';
+
+	/**
+	 * Checks whether we should trigger the event.
+	 */
+	public static function maybe_track() {
+		// UPE toggling is already being tracked. No need to trigger any event.
+		if ( get_option( self::IS_TRACKED_OPTION ) ) {
+			return;
+		}
+
+		// No need to track on new installs. We want the data
+		// from existing ones. New installs will have the events built-in.
+		if ( ! get_option( 'woocommerce_woocommerce_payments_version' ) ) {
+			self::mark_as_tracked();
+			return;
+		}
```

Comment:
Perhaps this isn't necessary as the feature flag won't be set on new installs, so the code below should serve this purpose. I'm also fine with merging as is, so feel free to ignore.

Reply:
Thanks for the feedback!
Right, makes sense. PR updated 🙂